docs/copilot/workspace-context.md

```diff
@@ -0,0 +1,75 @@
+- Making plans for complex code edits:
+  - `"@workspace how can I add a rich tooltip to a button?"` - Provides a plan for using the existing tooltip component with button elements
+  - `"@workspace add date validation to #selection"` - Plans how to apply the existing date validation to the selected code
+  - `"@workspace add a new API route for the forgot password form"` - Outlines where to add the new route and how to connect it to the existing code
+- Explaining higher-level concepts in a codebase:
+  - `"@workspace how is authentication implemented?"` - Overview of the authentication flow and references to the relevant code
+  - `"@workspace which API routes depend on this service?"` - Lists the routes that use the service in the selected code
+  - `"How do I build this #codebase?"` - List the steps to build the project based on documentation, scripts, and configurations
+
+## What sources does `@workspace` use for context?
+
+To answer your question, `@workspace` searches through the same sources a developer would use when navigating a codebase in VS Code:
+
+- All files in the workspace, except for files that are included in `.gitignore` files
+- Directory structure with nested folder and file names
+- GitHub's code search index, if the workspace is a GitHub repository and [indexed by code search](https://docs.github.com/en/enterprise-cloud@latest/copilot/github-copilot-enterprise/copilot-chat-in-github/using-github-copilot-chat-in-githubcom#asking-a-question-about-a-specific-repository-file-or-symbol)
+- Symbols and definitions in the workspace
+- Currently selected text or visible text in the active editor
+
+**Note**: `.gitignore` is bypassed if you have a file open or have text selected within an ignored file.
+
+## How does `@workspace` find the most relevant context
+
+Your full VS Code workspace can be too large to pass entirely to GitHub Copilot for responding to your chat prompt. Instead, `@workspace` extracts the most relevant information from the different context sources to ground Copilot's answer.
+
+First, `@workspace` determines which information is needed to answer your question, also including the conversation history, workspace structure, and currently selected code.
+
+Next, it performs a combination of search strategies, such as [GitHub's code search index](https://github.blog/2023-02-06-the-technology-behind-githubs-new-code-search), a lexical text searches over the local index to find local, uncommitted changes, and VS Code's language intelligence to add details like function signatures, parameters, and more.
+
+Finally, all of these pieces of context are ranked, sliced, and summarized by `@workspace`, and then used by GitHub Copilot to answer your question. The responses is marked up with references to files, file ranges, and symbols. This enables you to link directly from the chat response to the corresponding information in your codebase.
```

Comment:
I think we should keep this higher level. The first part is good, but for steps 2 and 3 I'd avoid talking about specifics like lexical searching or slicing, which may be confusing and also aren't really technically correct

Maybe something more like (needs word cleanup):

- First `@workspace` determines what context (file names, symbol names, code excepts, ...) from the workspace may help answer the question. 

- Next it collects the context using a variety of different approaches. This includes finding relevant snippets of code by searching locally or using Github Code search. We also collect context using VS Code's language IntelliSense's

- Finally we pass this context to copilot. If the context is large, we only pass the most potentially relevant parts of the context. The code snippets that were passed to copilot are listed as references in the response